src/Corax/Querying/Matches/SortingMatches/CompactKeyComparer.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using Sparrow;
+
+namespace Corax.Querying.Matches.SortingMatches;
+
+public static unsafe class CompactKeyComparer
+{
+    public static int Compare(UnmanagedSpan xItem, UnmanagedSpan yItem)
+    {
+        if (yItem.Address == null)
+        {
+            return xItem.Address == null ? 0 : 1;
+        }
+
+        if (xItem.Address == null)
+            return -1;
+        var match = Memory.Compare(xItem.Address + 1, yItem.Address + 1, Math.Min(xItem.Length - 1, yItem.Length - 1));
+        if (match != 0)
+            return match;
```

Comment:
Do we have to worry about the remainder in bits? 
So if we have Something that is 12 bits and another that is 11 bits?

Reply:
cc @redknightlois 
This was used previously in `ByTermComparer` and I just extracted it.
BTW: should we discard one character at the end?

Reply:
Same prefix, longer bits = larger key, so you have to assume the shorter key comes has zeroes at the end.